scripts/standby_node_health_check/standby_node_health_check.py

```diff
@@ -0,0 +1,450 @@
+#
+# Copyright © 2013-2024 NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+# This software product is a proprietary product of Nvidia Corporation and its affiliates
+# (the "Company") and all right, title, and interest in and to the software
+# product, including all associated intellectual property rights, are and
+# shall remain exclusively with the Company.
+#
+# This software product is governed by the End User License Agreement
+# provided with the software product.
+#
+
+import json
+import re
+import subprocess
+import argparse
+import sys
+from typing import List
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="Checking if standby node is ready.")
+
+    parser.add_argument(
+        "--fabric-interfaces",
+        nargs="+",
+        required=True,
+        help="Specify one or more fabric interfaces (at least one is required), eg ib0, mlx3_0",
+    )
+
+    parser.add_argument(
+        "--mgmt-interfaces",
+        nargs="+",
+        required=True,
+        help="Specify one or more management interfaces (at least one is required), eg eth0, eth1",
+    )
+
+    parser.add_argument("--version", action="version", version="%(prog)s 1.0.0")
+
+    args = parser.parse_args()
+    return args
+
+
+class StandbyNodeHealthChecker:
+    UFM_HA_CLUSTER_COMMAND = "ufm_ha_cluster {}"
+    SYSTEMCTL_IS_ACTIVE_COMMAND = "systemctl is-active {}.service"
+
+    def __init__(self, fabric_interfaces, mgmt_interfaces):
+        self._fabric_interfaces = fabric_interfaces
+        self._mgmt_interfaces = mgmt_interfaces
+        self._ufm_ha_status_string = ""
+
+    @classmethod
+    def _run_command(cls, command: str):
+        try:
+            result = subprocess.run(
+                command,
+                shell=True,
+                stdout=subprocess.PIPE,
+                universal_newlines=True,
+                check=False,
+            )
+            return result.returncode, result.stdout.replace("\t", "").strip()
+        except Exception as e:
+            return -1, "", str(e)
+
+    def _check_ib_interfaces(self):
+        result = True
+        ib_interfaces_status = self._run_and_parse_ibstat()
+        ib_to_ml_map = self._get_ib_to_mlx_port_mapping()
+        for ib_interface in self._fabric_interfaces:
+            ib_interface_to_validate = ib_interface
+            if not ib_interface.startswith("mlx"):
+                ib_interface_to_validate = ib_to_ml_map.get(ib_interface, ib_interface)
+            if ib_interface_to_validate not in ib_interfaces_status:
+                print(f"{ib_interface} is not in the list of IB interfaces")
+                result = False
+            elif not self._check_ib_interface(
+                ib_interface_to_validate, ib_interfaces_status
+            ):
+                print(
+                    f"IB interface {ib_interface} is not active "
+                    f"{ib_interfaces_status[ib_interface_to_validate]}"
```

Comment:
Not sure printing the dictionary is a user friendly message:
"_IB interface ib3 is not active {'State': 'down', 'Physical_state': 'polling'}_"

Reply:
Fixed